runner/operator/helix_filters/construct_helix_filters_input.py

```diff
@@ -0,0 +1,161 @@
+"""
+Main module that builds the JSON that needs to be submitted
+to the pipeline executor
+"""
+import logging
+import os
+import sys
+import json
+from runner.models import Port
+WORKDIR = os.path.dirname(os.path.abspath(__file__))
+LOGGER = logging.getLogger(__name__)
+
+
+def load_references():
+    """
+    Loads reference data from the resources JSON
+    """
+    data = json.load(open(os.path.join(WORKDIR, 'reference_jsons/helix_filters_resources.json'), 'rb'))
+    return data
+
+
+def get_baits_and_targets(assay, helix_filters_resources):
+    """
+    From value in assay, retrieve target files from helix_filters_resources
+    """
+    targets = helix_filters_resources['targets']
+
+    target_assay = assay
+
+    if assay.find("IMPACT410") > -1:
+        target_assay = "IMPACT410_b37"
+    if assay.find("IMPACT468") > -1:
+        target_assay = "IMPACT468_b37"
+    if assay.find("IMPACT341") > -1:
+        target_assay = "IMPACT341_b37"
+    if assay.find("IDT_Exome_v1_FP") > -1:
+        target_assay = "IDT_Exome_v1_FP_b37"
+    if assay.find("IMPACT468+08390") > -1:
+        target_assay = "IMPACT468_08390"
+    if assay.find("IMPACT468+Poirier_RB1_intron_V2") > -1:
+        target_assay = "IMPACT468_08050"
+
+    if target_assay in targets:
+        return {"class": "File", 'location': str(targets[target_assay]['targets_list'])}
+    else:
+        error_msg = "ERROR: Targets for Assay not found in helix_filters_resources.json: %s" % assay
+        LOGGER.error(error_msg)
+
+
+def create_cwl_file_obj(file_path):
+    """
+    Given a filepath, return a dictionary with class File and JUNO-specific URI
+    """
+    cwl_file_obj = {'class': 'File', 'location': "juno://%s" % file_path}
+    return cwl_file_obj
+
+
+def get_file_obj(file_obj):
+    """
+    Given file_obj, construct a dictionary of class File, that file's
+    JUNO-specific URI file path, and a list of secondary files with
+    JUNO-specific URI file paths
+    """
+    secondary_file_list = []
+    file_location = file_obj['location'].replace('file://', '')
```

Comment:
I think I mentioned this in some previous PR we should try to use `FileProcessor` for file manipulation. Most of the methods for extracting file paths from location etc is already implemented there.

Reply:
Ah, right. I copied this over from some other CWL